haskell/haskell.bzl

```diff
@@ -226,6 +227,42 @@ $ bazel-bin/.../hello-lib-repl  # run the script
 ```
 """
 
+def _haskell_prebuilt_library(ctx):
+  return [HaskellPrebuiltPackageInfo(
+            package = ctx.attr.package)]
+
+"""Wrap a prebuilt dependency as a rule.
+
+Example:
+  ```bzl
+  haskell_prebuilt_library(
+      name = "base_pkg",
+      package = "base",
+  )
+  haskell_library(
+      name = "hello-lib",
+      srcs = ["Lib.hs"],
+      deps = [
+          ":base",
+          "//hello-sublib:lib",
+      ],
+  )
+  ```
+
+This rule may wrap any prebuilt dependencies, i.e., GHC packages that aren't
+supplied by Bazel.  Depending on the wrapped rule eliminates the need to list
+the package name in prebuilt_dependencies.
+
+Often, rules of this type will be generated automatically by frameworks such
+as Hazel.
+"""
+haskell_prebuilt_library = rule(
+  _haskell_prebuilt_library,
+  attrs = dict(
+    package = attr.string(doc = "A non-Bazel-supplied GHC package name."),
```

Comment:
It would be good for the `package` to have `name` as the default value. In fact, could `name` alone be sufficient? The list of allowed characters in a package is a subset of the legal target names. Although people might want to import `ghc-prim` as `ghc_prim`.

Reply:
Good suggestion; I've set it default to name.  (I agree that `ghc-prim` vs `ghc_prim` means we should keep the separate attribute as an option.)